### PR TITLE
Handle jar file edge cases

### PR DESCRIPTION
**Pull request by tylerphelan**
CLI should be able to extract and upload:
- Jars with file and dir with same name: https://github.com/pivotal/kpack/issues/469
- FAR formatted JARs https://github.com/pivotal/kpack/issues/468

### CODE REVIEW

File: pkg/archive/zip.go [hunk 1]
```diff
@@ -79,3 +81,98 @@ func ExtractZip(src string, dest string) error {
...
+	var fileMode int64
+	for _, f := range zipReader.File {
+		fileMode = -1
+		if isFatFile(f.FileHeader) {
+			fileMode = 0777
+		}
+
+		var header *tar.Header
+		if f.Mode()&os.ModeSymlink != 0 {
+			target, err := func() (string, error) {
```
Comment (matthewmcnew) [inline pkg/archive/zip.go:110]: Could we extract these functions?

File: pkg/archive/zip.go [hunk 2]
```diff
@@ -79,3 +81,98 @@ func ExtractZip(src string, dest string) error {
...
+
+	var fileMode int64
+	for _, f := range zipReader.File {
+		fileMode = -1
+		if isFatFile(f.FileHeader) {
+			fileMode = 0777
+		}
+
+		var header *tar.Header
+		if f.Mode()&os.ModeSymlink != 0 {
```
Comment (matthewmcnew) [inline pkg/archive/zip.go:109]: Could this simplified with a golang switch?

**Comment by matthewmcnew** [comment]
I think this makes sense. I believe we still need to fix https://github.com/pivotal/kpack/issues/469 and https://github.com/pivotal/kpack/issues/468 inside kpack as well as the original issue resulted in problems with the blob source type.

**Review by matthewmcnew — APPROVED** [review]

File: pkg/archive/zip.go (hunk 2 above)
Comment (tylerphelan) [reply pkg/archive/zip.go:109]: I think since it's only one if/else it wouldn't improve it too much

**Review by djoyahoy — APPROVED** [review]

**Pull request merged**